src/init.cpp

```diff
@@ -59,9 +60,15 @@ template <>
 struct Init<Backend::MC> {
   static void initialize(const configuration& cfg) {
     memory::internal::initializeUmpireHostAllocator(cfg.umpire_host_memory_pool_initial_bytes);
+    if (pika::mpi::detail::environment::is_mpi_initialized()) {
+      pika::mpi::experimental::start_polling(pika::mpi::experimental::exception_mode::no_handler);
+    }
```

Comment:
Ideally, I'd have these removed (see https://github.com/pika-org/pika/pull/1267#discussion_r1804692945). If they stay I think this and the `stop_polling` should have the conditional removed:
```suggestion
      pika::mpi::experimental::start_polling(pika::mpi::experimental::exception_mode::no_handler);
```
There should be no situation where MPI isn't already initialized at this point (if anything we can add an assertion, but then it shouln't be calling `is_mpi_initialized` from pika's detail namespace, but something from DLA-Future or `MPI_Initialized` directly).

Reply:
unfortunately, there are some dlaf unit tests that don't use mpi and we need some kind of conditional check here

Reply:
Ah, yeah, makes sense. In that case, please don't use the internal pika API for checking if polling should be started.

Reply:
Never mind, I see you're not using `pika::mpi::detail::environment::is_mpi_initialized()` anymore. Thank you!